### PR TITLE
Make defaults off by default and some dependency updates

### PR DESCRIPTION
**Pull request by hzsweers**
This PR disables the defaults handling by default, as it imposes quite a bit of bloat (one method per property) onto users that may not actually need/want it.

This also opportunistically updates dependencies and to the new java-library format

### CODE REVIEW

**Comment by hzsweers** [comment]
Sidenote - updating gradle's helped me get around all the weird local build issues I'd had on previous PRs.

File: build.gradle [hunk 1]
```diff
@@ -20,3 +20,8 @@ allprojects {
         mavenCentral()
     }
 }
+
+task wrapper(type: Wrapper) {
+    gradleVersion = '4.0-rc-2'
```
Comment (serj-lotutovici) [inline build.gradle]: Gradle 4.0 is already released.

**Comment by bejibx** [comment]
But how to enable defaults generation now?

**Comment by hzsweers** [comment]
Use the compiler option

**Pull request merged**